assets/js/phoenix_live_view/view.js

```diff
@@ -875,7 +875,7 @@ export default class View {
     }
     this.pushWithReply(refGenerator, "event", event, resp => {
       DOM.showError(inputEl, this.liveSocket.binding(PHX_FEEDBACK_FOR))
-      if(DOM.isUploadInput(inputEl) && inputEl.getAttribute("data-phx-auto-upload") !== null){
+      if(DOM.isUploadInput(inputEl) && inputEl.dataset.phxAutoUpload !== null){
```

Comment:
i would either check for both undefined/null or do a less strict check which checks both 
```js
inputEl.dataset.phxAutoUpload !== null && inputEl.dataset.phxAutoUpload !== undefined
or
inputEl.dataset.phxAutoUpload != null
```

Reply:
> i would either check for both undefined/null or do a less strict check which checks both

That stops the reliance on the values being null or undefined, but I believe will also stop this pull request from fixing https://github.com/phoenixframework/phoenix_live_view/issues/2271 as it restores the original behaviour. The fix works because it changes the behaviour of the the check by passing and allowing upload when  inputEl.dataset.phxAutoUpload is undefined.